GeoLib/Polygon.cpp

```diff
@@ -191,44 +192,38 @@ bool Polygon::isPartOfPolylineInPolygon(const Polyline& ply) const
 			return true;
 		}
 	}
-	// check segment intersections
-	GeoLib::Point* s (new GeoLib::Point (0,0,0));
-	const std::size_t n_nodes(getNumberOfPoints() - 1);
-	for (std::size_t k(0); k < ply_size - 1; k++) {
-		for (std::size_t j(0); j < n_nodes; j++) {
-			if (GeoLib::lineSegmentIntersect(*(getPoint(j)), *(getPoint(j + 1)),
-							*(ply.getPoint(k)), *(ply.getPoint(k + 1)), *s)) {
-				delete s;
+
+	GeoLib::Point s;
+	for (auto polygon_seg : *this) {
+		for (auto polyline_seg : ply) {
+			if (GeoLib::lineSegmentIntersect(polyline_seg, polygon_seg, s)) {
 				return true;
 			}
 		}
 	}
 
-	delete s;
 	return false;
 }
 
-bool Polygon::getNextIntersectionPointPolygonLine (GeoLib::Point const & a,
-                GeoLib::Point const & b, GeoLib::Point* intersection_pnt,
-                std::size_t& seg_num) const
+bool Polygon::getNextIntersectionPointPolygonLine(
+    GeoLib::LineSegment const& seg, GeoLib::Point & intersection,
+    std::size_t& seg_num) const
 {
 	if (_simple_polygon_list.empty()) {
-		const std::size_t n_segments(getNumberOfPoints() - 1);
-		for (std::size_t k(seg_num); k < n_segments; k++) {
-			if (GeoLib::lineSegmentIntersect(*(getPoint(k)), *(getPoint(k + 1)), a, b, *intersection_pnt)) {
-				seg_num = k;
+		for (auto seg_it(begin()+seg_num); seg_it != end(); ++seg_it) {
```

Comment:
@chleh What do you mean by "wrap around at the last segment"?
By `seg_num != 0` it is possible to skip some segments. This can be useful when in a previous run the algorithm checked already `seg_num` line segments of the polyline.